scio-avro/src/test/scala/com/spotify/scio/avro/types/TypeProviderTest.scala

```diff
@@ -763,13 +754,45 @@ class TypeProviderTest extends FlatSpec with Matchers {
   class FromResource
 
   "AvroType.fromSchemaFile" should "support reading schema from multiline resource" in {
-    val r = FromResourceMultiLine(1)
-    r.test shouldBe 1
+    FromResourceMultiLine(1).test shouldBe 1
   }
 
   it should "support reading schema from resource" in {
-    val r = FromResource(2)
-    r.test shouldBe 2
+    FromResource(2).test shouldBe 2
+  }
+
+  class UserAnnot1 extends StaticAnnotation
+  class UserAnnot2 extends StaticAnnotation
+
+  def containsAllUserAnnotTypes[T: TypeTag]: Assertion =
+    typeOf[T].typeSymbol.annotations.map(_.tree.tpe)
+      .containsSlice(Seq(typeOf[UserAnnot1], typeOf[UserAnnot2])) shouldBe true
+
+  @UserAnnot1
+  @AvroType.fromSchemaFile("scio-avro/src/test/avro/scio-avro-test.avsc")
+  @UserAnnot2
+  class FromResourceWithUserAnnotations
+
+  it should "preserve user defined annotations" in {
+    containsAllUserAnnotTypes[FromResourceWithUserAnnotations]
   }
 
+  @UserAnnot1
+  @AvroType.fromSchema(
+    """{"type":"record","name": "Record","fields":[{"name":"f1","type":"int"}]}""")
+  @UserAnnot2
+  class SchemaWithUserAnnotations
+
+  "AvroType.fromSchema" should "preserve user defined annotations" in {
+    containsAllUserAnnotTypes[SchemaWithUserAnnotations]
+  }
+
+  @UserAnnot1
+  @AvroType.toSchema
+  @UserAnnot2
```

Comment:
can you test different orderings as well

Reply:
The length of the file exceeds the permissible limit. Where it is better to add tests with annotations?

Reply:
could be time to add another file to specifically test annotations

is it possible to increase the file size for tests @nevillelyh what do you think

Reply:
We can just add a pair of `scalastyle:{off,on}` to suppress the limit like this: https://github.com/spotify/scio/blob/master/scio-core/src/main/scala/com/spotify/scio/ScioContext.scala#L18